components/nexus-security/src/main/java/org/apache/shiro/nexus/ShiroJCacheAdapter.java

```diff
@@ -31,52 +33,36 @@
  * @since 3.0
  */
 public class ShiroJCacheAdapter<K, V>
+  extends ComponentSupport
   implements Cache<K, V>
 {
   private final javax.cache.Cache<K, V> cache;
 
+  private final String name;
+
   public ShiroJCacheAdapter(final javax.cache.Cache<K, V> cache) {
     this.cache = checkNotNull(cache);
+    this.name = cache.getName();
   }
 
   @Override
   public V get(final K key) throws CacheException {
-    try {
-      return cache.get(key);
-    }
-    catch (Throwable t) {
-      throw new CacheException(t);
-    }
+    return cache.get(key);
   }
 
   @Override
   public V put(final K key, final V value) throws CacheException {
-    try {
-      return cache.getAndPut(key, value);
-    }
-    catch (Throwable t) {
-      throw new CacheException(t);
-    }
+    return cache.getAndPut(key, value);
   }
 
   @Override
   public V remove(final K key) throws CacheException {
-    try {
-      return cache.getAndRemove(key);
-    }
-    catch (Throwable t) {
-      throw new CacheException(t);
-    }
+    return cache.getAndRemove(key);
   }
 
   @Override
   public void clear() throws CacheException {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove the declaration of thrown exception 'org.apache.shiro.cache.CacheException' which is a runtime exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)